lib/Sema/MiscDiagnostics.cpp

```diff
@@ -2084,14 +2084,10 @@ static void diagnoseUnownedImmediateDeallocationImpl(TypeChecker &TC,
     return;
 
   // Only diagnose for non-owning ownerships such as 'weak' and 'unowned'.
-  switch (ownershipAttr->get()) {
-  case ReferenceOwnership::Strong:
+  // Zero is the default/strong ownership strength.
+  if (ReferenceOwnership::Strong == ownershipAttr->get() ||
+      isLessStrongThan(ReferenceOwnership::Strong, ownershipAttr->get()))
```

Comment:
...are you expecting to add references that are stronger than strong?

Reply:
Yes: locks. (I'm working on an atomicity/reentrancy/concurrency model.)

Reply:
By "locks", do you mean an atomic strong reference (where a single variable can be accessed concurrently) or a reference that somehow guarantees exclusion on the referent object (where holding the reference somehow ensures that nobody else can modify the object)?

IIUC, the only thing using this strength-ordering is an optimization which, given two local references to the same object, strengthens the weaker one because it's known to stay valid because of the stronger reference.  That doesn't seem to be something you'd want for either definition: an atomic strong reference that's known not to be accessed externally should be strength-reduced to just an ordinary strong reference, rather than subsuming any such references; and an exclusive reference should presumably conflict with an attempt to form another reference to it (depending on your model?).